### PR TITLE
workflow: Add workflow to update CoC report on PR close event

### PR DESCRIPTION
**Pull request by nitin-ramchandani**
Added a new workflow to automate the process of updating a Pull Request (PR) record when a PR is closed. The workflow sends an HTTP request to the Code of Conduct (CoC) Handler, ensuring that the PR status is accurately reflected in our records.

### CODE REVIEW

File: .github/workflows/update-CoC-report.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: Update CoC report
+
+run-name: 'PR Closed: ${{ github.event.pull_request.number}} ${{ github.event.pull_request.title }}'
```
Comment (bjhargrave) [inline .github/workflows/update-CoC-report.yml:5]:
```suggestion
run-name: "${{ github.event.pull_request.title }} (#${{ github.event.pull_request.number}})"

```

File: .github/workflows/update-CoC-report.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  pull_request:
+    types: [ closed ]
```
Comment (bjhargrave) [inline .github/workflows/update-CoC-report.yml:8]:
```suggestion
    types:
      - closed
```

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: .github/workflows/update-CoC-report.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+jobs:
+  update-coc-insight-handler:
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (nitin-ramchandani) [reply .github/workflows/update-CoC-report.yml:23]: Correct, updated

**Comment by bjhargrave** [comment]
@mergifyio rebase

**Review by bjhargrave — APPROVED** [review]

**Pull request merged**